inc/class-cloner.php

```diff
@@ -1223,6 +1226,31 @@ protected function sameAsSource( $url ) {
 		return \Pressbooks\Utility\urls_have_same_host( $this->sourceBookUrl, $url );
 	}
 
+	/**
+	 * @param string $content
+	 *
+	 * @return $string
+	 */
+	protected function fixInternalLinks( $content ) {
+		// Fix relative URLs based on network type
+		if ( is_subdomain_install() ) {
+			// Remove book path if source book is in a subdirectory install and target book is in a subdomain install.
+			$content = preg_replace( '/href\="\/([a-z0-9]*)\/(front\-matter|chapter|back\-matter|part)/', 'href="/$2', $content );
+		} else {
+			// Get the directory path for the target book.
+			$subdirectory = $this->getSubdomainOrSubDirectory( $this->targetBookUrl );
+			// Update book path if source and target books are in subdirectory installs.
+			$content = preg_replace( '/href\="\/([a-z0-9]*)\/(front\-matter|chapter|back\-matter|part)/', 'href="/' . $subdirectory . '/$2', $content );
+			// Add book path if source book is in a subdomain install and target book is in a subdirectory install.
+			$content = preg_replace( '/href\="\/(front\-matter|chapter|back\-matter|part)/', 'href="/' . $subdirectory . '/$1', $content );
+		}
+
+		// Fix absolute URLs
+		$content = str_replace( 'href="' . untrailingslashit( $this->sourceBookUrl ), 'href="' . untrailingslashit( $this->targetBookUrl ), $content );
+
+		return $content;
+	}
+
```

Comment:
Per our discussion, I feel this would be better using the HTML5 parser looking for a href tags.

For example, If my books is on pressbooks.test/book1  how well does this handle me linking having both and internal link and a link to http://neuroscience.openetext.utoronto.ca/chapter/anatomy-physiology-the-cellular-levels-of-organization/ in the same chapter?

I feel this regex will touch edge cases it shouldn't.